scanner/src/main/kotlin/ArtifactoryCache.kt

```diff
@@ -52,44 +55,85 @@ class ArtifactoryCache(
+        val tempFile = createTempFile("scan-results-")
+
         try {
-            return OkHttpClientHelper.execute(Main.HTTP_CACHE_PATH, request).use { response ->
-                (response.code() == HttpURLConnection.HTTP_OK).also {
-                    val message = if (it) {
-                        response.body()?.let { body ->
-                            Okio.buffer(Okio.sink(target)).use { it.writeAll(body.source()) }
-                        }
+            OkHttpClientHelper.execute(Main.HTTP_CACHE_PATH, request).use { response ->
+                if (response.code() == HttpURLConnection.HTTP_OK) {
+                    response.body()?.let { body ->
+                        Okio.buffer(Okio.sink(tempFile)).use { it.writeAll(body.source()) }
+                    }
 
-                        if (response.cacheResponse() != null) {
-                            "Retrieved $cachePath from local cache."
-                        } else {
-                            "Downloaded $cachePath from Artifactory cache."
-                        }
+                    if (response.cacheResponse() != null) {
+                        log.info { "Retrieved $cachePath from local cache." }
                     } else {
+                        log.info { "Downloaded $cachePath from Artifactory cache." }
+                    }
+
+                    return yamlMapper.readValue(tempFile, ScanResults::class.java)
+                } else {
+                    log.info {
                         "Could not get $cachePath from Artifactory cache: ${response.code()} - " +
                                 response.message()
                     }
-
-                    log.info { message }
                 }
             }
         } catch (e: IOException) {
             e.showStackTrace()
 
             log.warn { "Could not get $cachePath from Artifactory cache: ${e.message}" }
+        }
 
-            return false
+        return ScanResults(id, emptyList())
+    }
+
+    override fun read(id: Identifier, scannerSpecification: ScannerSpecification): ScanResults {
+        val scanResults = read(id)
+        val filteredResults = scanResults.results.filter {
+            it.scanner == scannerSpecification
+        }
+
+        if (filteredResults.isEmpty() && scanResults.results.isNotEmpty()) {
+            log.info { "No cached scan results found for '$scannerSpecification', ignoring cached scan results for " +
+                    "scanners: ${scanResults.results.map { it.scanner }}" }
         }
+
+        return ScanResults(id, filteredResults)
     }
 
-    override fun write(pkg: Package, source: File): Boolean {
-        val cachePath = cachePath(pkg, source)
+    override fun add(id: Identifier, scanResult: ScanResult): Boolean {
+        if (scanResult.summary.fileCount == 0) {
+            // Do not cache empty scan results. It is likely that something went wrong when they were created, and if
+            // not, it is cheap to re-create them.
+
```

Comment:
Nit: Remove blank line (compare to below).

Reply:
Below there is also an empty line. I added the empty line to make it more visible that the comment does not belong to the line below, but to the whole block. Maybe it should be above the `if` instead?

Reply:
I don't care strongly either way, I just wanted to have it consistent in both `if`s. Sorry if I mis-spotted this and it already is consistent.